src/isar/storage/uploader.py

```diff
@@ -0,0 +1,36 @@
+import logging
+from queue import Empty, Queue
+from typing import List
+
+from isar.models.mission_metadata.mission_metadata import MissionMetadata
+from isar.storage.storage_interface import StorageException, StorageInterface
+from robot_interface.models.inspection.inspection import Inspection
+
+
+class Uploader:
+    def __init__(
+        self, upload_queue: Queue, storage_handlers: List[StorageInterface]
+    ) -> None:
+        self.upload_queue: Queue = upload_queue
+        self.storage_handlers: List[StorageInterface] = storage_handlers
+
+        self.logger = logging.getLogger("uploader")
+
+    def run(self) -> None:
+        while True:
+            inspection: Inspection
+            mission_metadata: MissionMetadata
+            try:
+                inspection, mission_metadata = self.upload_queue.get(timeout=5)
```

Comment:
Should the timeout be hardcoded here? Also, is 5 sufficient for the uploader? If there is a larger inspection, for instance a video, could we expect it to need more time?

Reply:
I don't see a reason to have this as a configuration variable. It could almost be argued that the queue could be blocking in this thread although I haven't gone for that approach as we will be adding additional functionality to handle back-off uploads of those that may have failed previously. 

The timeout could have been any value in principle.